altair/utils/schemapi.py

```diff
@@ -247,82 +495,57 @@ def split_into_equal_parts(n, p):
                 # Insert newlines and spacing after the last element in each row
                 if num == (len(param_names_row) - 1):
                     param_names_table += "\n"
-                    # 16 is the indendation of the returned multiline string below
-                    param_names_table += " " * 16
         return param_names_table
 
-    def __str__(self):
-        # Try to get the lowest class possible in the chart hierarchy so
-        # it can be displayed in the error message. This should lead to more informative
-        # error messages pointing the user closer to the source of the issue.
-        for prop_name in reversed(self.absolute_path):
-            # Check if str as e.g. first item can be a 0
-            if isinstance(prop_name, str):
-                potential_class_name = prop_name[0].upper() + prop_name[1:]
-                cls = getattr(vegalite, potential_class_name, None)
-                if cls is not None:
-                    break
-        else:
-            # Did not find a suitable class based on traversing the path so we fall
-            # back on the class of the top-level object which created
-            # the SchemaValidationError
-            cls = self.obj.__class__
-
-        # Output all existing parameters when an unknown parameter is specified
-        if self.validator == "additionalProperties":
-            param_dict_keys = inspect.signature(cls).parameters.keys()
-            param_names_table = self._format_params_as_table(param_dict_keys)
-
-            # `cleandoc` removes multiline string indentation in the output
-            return inspect.cleandoc(
-                """`{}` has no parameter named {!r}
-
-                Existing parameter names are:
-                {}
-                See the help for `{}` to read the full description of these parameters
-                """.format(
-                    cls.__name__,
-                    self.message.split("('")[-1].split("'")[0],
-                    param_names_table,
-                    cls.__name__,
-                )
-            )
-        # Use the default error message for all other cases than unknown parameter errors
+    def _get_default_error_message(
+        self,
+        errors: ValidationErrorList,
+    ) -> str:
+        bullet_points: List[str] = []
+        errors_by_validator = _group_errors_by_validator(errors)
+        if "enum" in errors_by_validator:
+            for error in errors_by_validator["enum"]:
+                bullet_points.append(f"one of {error.validator_value}")
+
+        if "type" in errors_by_validator:
+            types = [f"'{err.validator_value}'" for err in errors_by_validator["type"]]
+            point = "of type "
```

Comment:
```suggestion
                bullet_points.append(f"One of {error.validator_value}")

        if "type" in errors_by_validator:
            types = [f"'{err.validator_value}'" for err in errors_by_validator["type"]]
            point = "Of type "
```

```suggestion
                bullet_points.append(f"One of {error.validator_value}")
```

What do you think of capitalizing these? So instead of the current version:

```
SchemaValidationError: 'asdf' is an invalid value for `stack`. Valid values are:

- one of ['zero', 'center', 'normalize']
- of type 'null' or 'boolean'
```

it would be:

```
SchemaValidationError: 'asdf' is an invalid value for `stack`. Valid values are:

- One of ['zero', 'center', 'normalize']
- Of type 'null' or 'boolean'
```

I don't feel strongly here, just a suggestion. I'm thinking this could help the bullet point sentences appear more separate from each other, but maybe it is just personal taste

Reply:
I like it! Included in the latest commits